accepted/2020/or-greater-defines/or-greater-defines.md

```diff
@@ -0,0 +1,221 @@
+      `NET48_OR_GREATER`, `NET472_OR_GREATER`..`NET20_OR_GREATER`.
+* .NET Core
+    - Applies to `netcoreappX.Y` and the `NETCOREAPP` symbol
+    - For example, `netcoreapp3.1` will define `NETCOREAPP`, `NETCOREAPP3_1`,
+      `NETCOREAPP3_1_OR_GREATER`..`NETCOREAPP1_0_OR_GREATER`.
+* .NET 5 and later
+    - Applies to `netX.Y` and the `NET` symbol
+    - For example, `net6.0` will define `NET`, `NET6_0`, `NET6_0_OR_GREATER` and
+      `NET5_0_OR_GREATER`.
+    - This will also include the corresponding defines a .NET Core 3.1 successor
+      would have gotten. For example, `net5.0` will also define `NETCOREAPP`,
+      `NETCOREAPP3_1_OR_GREATER`..`NETCOREAPP1_0_OR_GREATER`.
+    - This will *neither* define a `NETCOREAPP5_0` nor
+      `NETCOREAPP5_0_OR_GREATER`.
+* .NET 5 and later with operating systems
+    - The OS flavors will also gain the `XXX_OR_GREATER` variants
+    - For example, `net5.0-windows10.0.19222.0` will also define `WINDOWS`,
+      `WINDOWS10_0_19041_0`, `WINDOWS10_0_19041_0_OR_GREATER`, etc.
+* Xamarin
+    - This covers the existing Xamarin offerings, the new .NET 6-based iOS and
+      Android support is handled by the previous sections.
+    - The existing Xamarin Apple platforms don't have versioned preprocessor
+      symbols, but Android does: `__ANDROID_16__`
+    - We don't plan on adding `OR_GREATER` flavors like
+      `__ANDROID_16_OR_GREATER__`. Instead, we're only doing those for .NET 6.
+
+### Doc changes
+
+The [documentation table][docs-table] in the [#if (C# reference)][docs] will
+need to change as follows:
+
+> | Target Frameworks | Symbols |
+> | ------------------| ------- |
+> | .NET Framework    | `NETFRAMEWORK`, `NET48`, `NET472`, `NET471`, `NET47`, `NET462`, `NET461`, `NET46`, `NET452`, `NET451`, `NET45`, `NET40`, `NET35`, `NET20`, `NET48_OR_GREATER`, `NET472_OR_GREATER`, `NET471_OR_GREATER`, `NET47_OR_GREATER`, `NET462_OR_GREATER`, `NET461_OR_GREATER`, `NET46_OR_GREATER`, `NET452_OR_GREATER`, `NET451_OR_GREATER`, `NET45_OR_GREATER`, `NET40_OR_GREATER`, `NET35_OR_GREATER`, `NET20_OR_GREATER` |
+> | .NET Standard     | `NETSTANDARD`, `NETSTANDARD2_1`, `NETSTANDARD2_0`, `NETSTANDARD1_6`, `NETSTANDARD1_5`, `NETSTANDARD1_4`, `NETSTANDARD1_3`, `NETSTANDARD1_2`, `NETSTANDARD1_1`, `NETSTANDARD1_0`, `NETSTANDARD2_1_OR_GREATER`, `NETSTANDARD2_0_OR_GREATER`, `NETSTANDARD1_6_OR_GREATER`, `NETSTANDARD1_5_OR_GREATER`, `NETSTANDARD1_4_OR_GREATER`, `NETSTANDARD1_3_OR_GREATER`, `NETSTANDARD1_2_OR_GREATER`, `NETSTANDARD1_1_OR_GREATER`, `NETSTANDARD1_0_OR_GREATER` |
+> | .NET 6 (and .NET Core) | `NET5_0`, `NETCOREAPP`, `NETCOREAPP3_1`, `NETCOREAPP3_0`, `NETCOREAPP2_2`, `NETCOREAPP2_1`, `NETCOREAPP2_0`, `NETCOREAPP1_1`, `NETCOREAPP1_0`, `NET6_0_OR_GREATER`, `NET5_0_OR_GREATER`, `NETCOREAPP3_1_OR_GREATER`, `NETCOREAPP3_0_OR_GREATER`, `NETCOREAPP2_2_OR_GREATER`, `NETCOREAPP2_1_OR_GREATER`, `NETCOREAPP2_0_OR_GREATER`, `NETCOREAPP1_1_OR_GREATER`, `NETCOREAPP1_0_OR_GREATER` |
```

Comment:
I wonder if we can leave off the `_0` from .NET5 and later, since we don't plan on releasing minor versions?

It'd save a few characters to type: `NET5_OR_GREATER`.

Reply:
That probably goes hand in hand with [a discussion on Twitter](https://twitter.com/jaredpar/status/1324004651948150784) that happened a few days ago. Basically, `NET40_OR_GREATER` would probably conflict with what `NET40` refers to right now.

Reply:
ah, right, I didn't see that .NET Framework symbols use the shortened name so they'd conflict.